expr/helper/align.go

```diff
@@ -33,25 +34,50 @@ func LCM(args ...int64) int64 {
 	return lcm
 }
 
-// GetCommonStep returns LCM(steps) for slice of metrics.
-// minStart and maxStop will be set to closest lower or equal multiple of LCM(steps).
-func GetCommonStep(args []*types.MetricData) int64 {
+// GetCommonStep returns LCM(steps), changed (bool) for slice of metrics.
+// If all metrics has the same step, changed == false.
+func GetCommonStep(args []*types.MetricData) (commonStep int64, changed bool) {
+	changed = false
 	steps := make([]int64, 0, len(args))
+	firstStep := args[0].StepTime
 	for _, arg := range args {
 		steps = append(steps, arg.StepTime)
+		if !changed && firstStep != arg.StepTime {
```

Comment:
It makes sense to check if a step is already in steps. Will see, how to do it in faster way